### PR TITLE
Fix compiler warnings

### PR DESCRIPTION
**Pull request by 4tXJ7f**
This commit fixes two compiler warnings related to parentheses and two
warnings related to derived classes hiding methods from the base class
with the same name.

### CODE REVIEW

File: src/theory/quantifiers/sygus/cegis_unif.h [hunk 1]
```diff
@@ -47,6 +47,10 @@ namespace quantifiers {
 class CegisUnifEnumDecisionStrategy : public DecisionStrategyFmf
 {
  public:
+  // Do not hide the zero-argument version of initialize() inherited from the
```
Comment (ajreynol) [inline src/theory/quantifiers/sygus/cegis_unif.h]: Style question here: should this go beneath the constructor? Also, the comment maybe should be /** */?

File: src/theory/strings/theory_strings.h [hunk 2]
```diff
@@ -728,6 +728,10 @@ class TheoryStrings : public Theory {
   class StringSumLengthDecisionStrategy : public DecisionStrategyFmf
   {
    public:
+    // Do not hide the zero-argument version of initialize() inherited from the
```
Comment (ajreynol) [inline src/theory/strings/theory_strings.h]: Ditto

File: src/theory/quantifiers/sygus/cegis_unif.h (hunk 1 above)
Comment (4tXJ7f) [reply src/theory/quantifiers/sygus/cegis_unif.h]: Are you talking about moving the comment or the `using` line as well? Maybe moving it next to the other version of `initialize()` would be a good idea. The `/** */` comments are actually Doxygen comments. Maybe I can use `/* */` though.

**Comment by ajreynol** [comment]
I’m talking about moving the using line as well.
And yes I think I’d prefer having a /* */ comment.

**Comment by 4tXJ7f** [comment]
Alright, I addressed the comments, PTAL.

**Pull request merged**